src/ir/context.rs

```diff
@@ -964,6 +965,29 @@ impl BindgenContext {
         }
     }
 
+    /// Temporarily loan `Item` with the given `ItemId`. This provides means to
+    /// mutably borrow `Item` while having a reference to `BindgenContext`.
+    ///
+    /// `Item` with the given `ItemId` is removed from the context, given
+    /// closure is executed and then `Item` is placed back.
+    ///
+    /// # Panics
+    ///
+    /// Panics if attempt to resolve given `ItemId` inside the given
+    /// closure is made.
+    fn with_loaned_item<F: FnOnce(&BindgenContext, &mut Item)>(
```

Comment:
For legibility, let's move the `FnOnce` bound to a `where` clause. Additionally, we should allow the function to return things.

```rust
fn with_loaned_item<F, T>(&mut self, id: ItemId, f: F) -> T
where
    F: FnOnce(&BindgenContext, &mut Item) -> T
{
    ...
}
```

Reply:
Yeah, I agree, it is more legible. But `rustfmt` thinks otherwise. 
Is `rustfmt` used to move bounds into `where` in the past?

> Additionally, we should allow the function to return things.

Good point!